bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/XmlDocumentBundleTracker.java

```diff
@@ -34,27 +48,26 @@
  * @author Michael Grammling - Initial Contribution
  * @author Benedikt Niehues - Changed resource handling so that resources can be
  *         patched by fragments.
+ * @author Simon Kaufmann - Tracking of remaining bundles
  *
  * @param <T>
  *            the result type of the conversion
  */
+@SuppressWarnings({ "unchecked", "rawtypes" })
```

Comment:
Using this changes we don't need to supress warnings (perhaps `?` could be changed to some other type, I just used a q'n'd fix).

```diff
diff --git a/bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/XmlDocumentBundleTracker.java b/bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/XmlDocumentBundleTracker.java
index c9a97abb6..cc04466d3 100644
--- a/bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/XmlDocumentBundleTracker.java
+++ b/bundles/config/org.eclipse.smarthome.config.xml/src/main/java/org/eclipse/smarthome/config/xml/osgi/XmlDocumentBundleTracker.java
@@ -53,7 +53,6 @@ import org.slf4j.LoggerFactory;
  * @param <T>
  *            the result type of the conversion
  */
-@SuppressWarnings({ "unchecked", "rawtypes" })
 public class XmlDocumentBundleTracker<T> extends BundleTracker<Bundle> {
 
     private final Logger logger = LoggerFactory.getLogger(XmlDocumentBundleTracker.class);
@@ -64,10 +63,10 @@ public class XmlDocumentBundleTracker<T> extends BundleTracker<Bundle> {
     private final Map<Bundle, XmlDocumentProvider<T>> bundleDocumentProviderMap = new ConcurrentHashMap<>();
     private final Map<Bundle, ScheduledFuture<?>> queue = new ConcurrentHashMap<>();
     private final Set<Bundle> finishedBundles = new CopyOnWriteArraySet<>();
-    private final Map<String, ServiceRegistration> bundleReadyMarkerRegistrations = new ConcurrentHashMap<>();
+    private final Map<String, ServiceRegistration<?>> bundleReadyMarkerRegistrations = new ConcurrentHashMap<>();
     private final String readyMarkerKey;
 
-    private BundleTracker relevantBundlesTracker;
+    private BundleTracker<?> relevantBundlesTracker;
 
     /**
      * Creates a new instance of this class with the specified parameters.
@@ -125,7 +124,7 @@ public class XmlDocumentBundleTracker<T> extends BundleTracker<Bundle> {
 
     @Override
     public final synchronized void open() {
-        relevantBundlesTracker = new BundleTracker(context,
+        relevantBundlesTracker = new BundleTracker<Object>(context,
                 Bundle.RESOLVED | Bundle.STARTING | Bundle.STOPPING | Bundle.ACTIVE, null) {
             @Override
             public Object addingBundle(Bundle bundle, BundleEvent event) {
@@ -342,21 +341,21 @@ public class XmlDocumentBundleTracker<T> extends BundleTracker<Bundle> {
     private void registerReadyMarker(Bundle bundle) {
         String bsn = bundle.getSymbolicName();
         if (!bundleReadyMarkerRegistrations.containsKey(bsn)) {
-            ServiceRegistration reg = ReadyUtil.markAsReady(context, readyMarkerKey, bsn);
+            ServiceRegistration<?> reg = ReadyUtil.markAsReady(context, readyMarkerKey, bsn);
             bundleReadyMarkerRegistrations.put(bsn, reg);
         }
     }
 
     private void unregisterReadyMarker(Bundle bundle) {
         String bsn = bundle.getSymbolicName();
-        ServiceRegistration reg = bundleReadyMarkerRegistrations.remove(bsn);
+        ServiceRegistration<?> reg = bundleReadyMarkerRegistrations.remove(bsn);
         if (reg != null) {
             reg.unregister();
         }
     }
 
     private void unregisterReadyMarkers() {
-        for (ServiceRegistration reg : bundleReadyMarkerRegistrations.values()) {
+        for (ServiceRegistration<?> reg : bundleReadyMarkerRegistrations.values()) {
             reg.unregister();
         }
         bundleReadyMarkerRegistrations.clear();
```